prog/dftb+/lib_type/typegeometryhsd.F90

```diff
@@ -317,4 +318,129 @@ subroutine readTGeometryGen_help(node, geo, text)
 
   end subroutine readTGeometryGen_help
 
+
+  !> Reads the geometry from a node in a HSD tree in XYZ format
+  subroutine readTGeometryXyz(node, geo)
+
+    !> Node containing the geometry in XYZ format
+    type(fnode), pointer :: node
+
+    !> Contains the geometry on exit
+    type(TGeometry), intent(out) :: geo
+
+    type(string) :: text
+
+    call getFirstTextChild(node, text)
+    call readTGeometryXyz_help(node, geo, char(text))
+
+  end subroutine readTGeometryXyz
+
+
+  !> Helping routine for reading geometry from a HSD tree in GEN format
+  subroutine readTGeometryXyz_help(node, geo, text)
+
+    !> Node to parse (only needed to produce proper error messages)
+    type(fnode), pointer :: node
+
+    !> Contains the geometry on exit
+    type(TGeometry), intent(out) :: geo
+
+    !> Text content of the node
+    character(len=*), intent(in) :: text
+
+    type(string) :: txt
+    integer :: iStart, iOldStart, iErr, iEnd
+    integer :: ii, iSp, iTmp
+    real(dp) :: coords(3), rTmp, det
+    type(listString) :: speciesNames
+
+    ! Read first line of the xyz file: Number of atoms
+    iStart = 1
+    iEnd = nextLine(text, iStart)
+    call getNextToken(text(:iEnd), geo%nAtom, iStart, iErr)
+    call checkError(node, iErr, "Bad number of atoms.")
+    ! xyz files are always molecular, and we are not going to support extensions to this
```

Comment:
The jmol variant format does support lattice vectors (probably several other hacked implementations as well), so that might be for a future extension to this routine.

Reply:
My problem with all this flavours of xyz is that they are not well standardized. And usually you want a molecular calculation if you have a xyz file, so I don't see much use cases here.